src/sgx/SGXWAMRWasmModule.cpp

```diff
@@ -86,6 +87,11 @@ void SGXWAMRWasmModule::doBindToFunction(faabric::Message& msg, bool cache)
                      faasmSgxErrorString(returnValue));
         throw std::runtime_error("Unable to load WASM module");
     }
+
+    // Set up the thread stacks
+    // 28/06/2021 - Threading is not supported in SGX-WAMR. Thus, we don't care
+    // what we put in this vector.
+    threadStacks.push_back(-1);
```

Comment:
Can it be left empty or is there a check that it's the right size? Might be good to say why we have to put a value in there if we don't care about it.

Reply:
Yes, I also hesitated on this one. There's an assertion [here](https://github.com/faasm/faasm/blob/fe5cbb30c7add9d139f5fa7c84dc2e0fa0ed7d7b/src/wasm/WasmModule.cpp#L341) that makes us have to add at least one element.

I feel the assertion must stay, as this thread management is something we'd desirably have in all our runtimes. Thus it makes more sense to deliberately override it for SGX. In an ideal scenario, we'd support threading in SGX as well.

Reply:
What exactly does us stop from supporting multithreading in WAMR in SGX? Does WAMR-SGX itself lack support?